components/script/dom/idbrequest.rs

```diff
@@ -0,0 +1,277 @@
+        self.ready_state.set(IDBRequestReadyState::Done);
+    }
+
+    pub fn set_result(&self, result: HandleValue) {
+        self.result.set(result.get());
+    }
+
+    pub fn set_error(&self, error: Error) {
+        // FIXME:(rasviitanen) Support all error types
+        if let Error::Version = error {
+            self.error.set(Some(&DOMException::new(
+                &self.global(),
+                DOMErrorName::VersionError,
+            )));
+        }
+    }
+
+    pub fn set_transaction(&self, transaction: &IDBTransaction) {
+        self.transaction.set(Some(transaction));
+    }
+
+    // https://www.w3.org/TR/IndexedDB-2/#asynchronously-execute-a-request
+    pub fn execute_async(
+        source: &IDBObjectStore,
+        operation: AsyncOperation,
+        request: Option<DomRoot<IDBRequest>>,
+    ) -> Fallible<DomRoot<IDBRequest>> {
+        // Step 1: Let transaction be the transaction associated with source.
+        let transaction = source.transaction().expect("Store has no transaction");
+        let global = transaction.global();
+
+        // Step 2: Assert: transaction is active.
+        if !transaction.is_active() {
+            return Err(Error::TransactionInactive);
+        }
+
+        // Step 3: If request was not given, let request be a new request with source as source.
+        let request = request.unwrap_or({
+            let new_request = IDBRequest::new(&global);
+            new_request.set_source(Some(source));
+            new_request.set_transaction(&transaction);
+            new_request
+        });
+
+        // Step 4: Add request to the end of transaction’s request list.
+        transaction.add_request(&request);
+
+        // Step 5: Run the operation, and queue a returning task in parallel
+        // the result will be put into `receiver`
+        let transaction_mode = match transaction.get_mode() {
+            IDBTransactionMode::Readonly => IndexedDBTxnMode::Readonly,
+            IDBTransactionMode::Readwrite => IndexedDBTxnMode::Readwrite,
+            IDBTransactionMode::Versionchange => IndexedDBTxnMode::Versionchange,
+        };
+
+        let (sender, receiver) = ipc::channel(global.time_profiler_chan().clone()).unwrap();
+
+        let response_listener = RequestListener {
+            request: Trusted::new(&request),
+        };
+
+        let task_source = global.database_access_task_source();
+        let trusted_global = Trusted::new(&*global);
+        ROUTER.add_route(
+            receiver.to_opaque(),
+            Box::new(move |message| {
+                let response_listener = response_listener.clone();
+                let _ = task_source.queue(
+                    task!(request_callback: move || {
+                        response_listener.handle_async_request_finished(
+                            message.to().expect("Could not unwrap message"));
+                    }),
+                    trusted_global.root().upcast(),
```

Comment:
Same issue here.

Reply:
yeah, I should probably get some sleep. I'm missing some really obvious things right now. I even thought I compiled and tested the program with `mach check`... Anyway, I've fixed this now.